nixos/modules/system/boot/loader/systemd-boot/systemd-boot.nix

```diff
@@ -94,6 +98,17 @@ in {
       '';
     };
 
+    xbootldrMountPoint = mkOption {
+      default = null;
+      type = types.nullOr types.str;
+      description = lib.mdDoc ''
+        Where the XBOOTLOADER partition is mounted.
+
+        If set, this partition will be used as $BOOT to store boot loader entries and extra files
+        instead of the ESP.
+      '';
+    };
+
```

Comment:
I am thinking, maybe an assert to check that this mountPoint is actually define in `fileSystems` would help prevent mistakes here.

Reply:
I don't know if we care about this, but some people do allow `systemd-gpt-auto-generator` to create their `/boot` mount (and with this PR, they could do both their `/efi` and `/boot` mounts), meaning they wouldn't be in `fileSystems` at all.

Reply:
I happen to use `systemd-gpt-auto-generator` as a matter of fact, so checking `fileSystems` won't work.

It's true that there is no reverting back if the systemd-boot config is messed up. I can setup an elaborate test in the python script to see if  `xbootldrMountPoint`/`efiSysMountPoint` are mountpoints and if the partition UUID is as expected, but that will need time.

Is such a test a blocker for the PR?

Reply:
I think it should be a blocker yes. As it is, an uncaring user might end up making their machine unbootable, we should have guards against that. Using an assertion doesn't seem to be the good implementation if there is a `systemd-gpt-auto-generator` usecase but something in the backend should be possible.